Update options + symfony specific integrations doc (#14891)
getsentry/sentry-docs: docs/platforms/php/guides/symfony/integrations/index.mdx

```php
// The amount of reserved memory every request (in bytes), this has to be a positive integer amount of bytes. Defaults to 16 KiB
$errorHandler = \Sentry\ErrorHandler::registerOnceFatalErrorHandler(16 * 1024);
// The amount of memory in bytes to increase the `memory_limit` to when handling a out of memory error, can also be set to `null` to disable increasing the `memory_limit`. Defaults to 5 MiB
$errorHandler->setMemoryLimitIncreaseOnOutOfMemoryErrorInBytes(5 * 1024 * 1024);
```

For some frameworks or projects, specific integrations are provided both officially and by third-party users that automatically register the error handlers. Please refer to their documentation.

By default, the error*types option defaults to the value returned by the `error_reporting()` function, which can change at runtime. Alternately, you can set the option to a fixed value by setting its value to a bitmask of the PHP `E*\*`constants in`\Sentry\init()`. In this case, Sentry will log only errors of those specific types regardless of the current reporting level.

### FatalErrorListenerIntegration

This integration catches all fatal errors and emits the corresponding events.

> To do so, it ensures that Sentry's `ErrorHandler` is registered, and adds a callback to it as a fatal error listener.

Reviewer: ```suggestion
It does this by ensuring that Sentry's `ErrorHandler` is registered, and adds a callback to it as a fatal error listener.
```

This integration reads excerpts of code around the line that originated an error.

### EnvironmentIntegration

This integration fills the event data with PHP runtime and server OS information.

### ModulesIntegration

This integration logs all the versions of the packages installed with Composer with the event details; the root project is also included.

## Custom Integrations

You can customize the list of integrations without disabling the defaults by using the `options.integrations` setting in Symfony. Provide service IDs for classes implementing `\Sentry\Integration\IntegrationInterface`.

```yaml {filename:config/packages/sentry.yaml}
sentry:
    options:
        integrations:
            - "sentry.integration.my_custom_integration"
```

Register your integration as a service:

```yaml {filename:config/services.yaml}
services:
    sentry.integration.my_custom_integration:
        class: App\Sentry\Integration\MyCustomIntegration
```

If you want to remove one or more default integrations, set `options.default_integrations: false` and then explicitly list the integrations you want to enable via `options.integrations`.

### Use a Callback to Customize Integrations

> For advanced cases, you can provide a callback (service) instead of a fixed list to customize integrations. See how to reference callables in Symfony configuration here: <PlatformLink to="/platforms/php/guides/symfony/configuration/symfony-options/#callables">Callables in Symfony Options</PlatformLink>.

Reviewer: This link does not exist. The closest I could find was this: `/platforms/php/guides/symfony/configuration/options/`, however I could not find the `#callables` option. Is that in a separate PR?
